apis/v1beta1/providerconfig_types.go

```diff
@@ -26,6 +26,10 @@ import (
 type ProviderConfigSpec struct {
 	// Credentials required to authenticate to this provider.
 	Credentials ProviderCredentials `json:"credentials"`
+
+	// Endpoint is where you can override the default endpoint configuration
+	// of AWS calls made by the provider.
+	Endpoint *EndpointConfig `json:"endpoint,omitempty"`
```

Comment:
nitpick: (and on other `omitempty` fields below)
```suggestion
	// of AWS calls made by the provider.
	// +optional
	Endpoint *EndpointConfig `json:"endpoint,omitempty"`
```